core/src/test/java/apoc/spatial/GeocodeTest.java

```diff
@@ -287,6 +297,23 @@ private void testGeocodeAddress(String address, double lat, double lon, Map<Stri
                     } else {
                         // over request limit
                     }
+                    return null;
                 });
     }
+
+    private void waitForServerResponseOK(String query, Map<String, Object> params, ResultTransformer<Object> objectResultTransformer) {
+        assertEventually(() -> {
+            try {
+                db.executeTransactionally(query,
+                        params,
+                        objectResultTransformer);
+                return true;
+            } catch (Exception e) {
+                if (e.getMessage().contains("Server returned HTTP response code")) {
+                    return false;
+                }
+                throw e;
+            }
+        }, (value) -> value, 20L, TimeUnit.SECONDS);
```

Comment:
Just want to make sure I understood this correctly, this will retry to call the procedure if it gets a 502 etc until it gets a 200 for max 20 seconds? otherwise it will throw an error? :)

Reply:
I took a look at the failing tests, and both are to do with geocode, one is failing due to a 502 ;) might wanna check those out as well :)